resource_customizations/numaflow.numaproj.io/MonoVertex/actions/discovery.lua

```diff
@@ -18,7 +25,18 @@ if obj.spec.lifecycle ~= nil and obj.spec.lifecycle.desiredPhase ~= nil and obj.
   paused = true
 end
 if paused then
-  actions["unpause"]["disabled"] = false
+  if obj.spec.metadata ~= nil and  obj.spec.metadata.annotations ~= nil and obj.spec.metadata.annotations["numaflow.numaproj.io/allowed-resume-strategies"] ~= nil then
+    if obj.spec.metadata.annotations["numaflow.numaproj.io/allowed-resume-strategies"] == "fast" then
+      actions["unpause-fast"]["disabled"] = false
+    elseif obj.spec.metadata.annotations["numaflow.numaproj.io/allowed-resume-strategies"] == "slow, fast" then
+      actions["unpause-gradual"]["disabled"] = false
+      actions["unpause-fast"]["disabled"] = false
+    else
+      actions["unpause-gradual"]["disabled"] = false
```

Comment:
This is probably okay, but consider maybe we could use something like [find() function](https://www.codecademy.com/resources/docs/lua/strings/find) to find "slow" or to find "fast" in the string. 

I realize that for us, we will have the UI literally write "slow, fast", so it isn't like we necessarily need to allow "fast, slow" as well or anything.

I am okay if we keep as is, though.

Reply:
In any case, can you add comments to explain the purpose of this code?